src/textbox.cpp

```diff
@@ -250,7 +250,28 @@ bool TextBox::mouseButtonEvent(const Vector2i &p, int button, bool down,
                 /* Double-click: select all text */
                 mSelectionPos = 0;
                 mCursorPos = (int) mValueTemp.size();
-                mMouseDownPos = Vector2i(-1, 1);
```

Comment:
For the first question, I found it necessary to add this otherwise I could
not drag with the right click. I did not want to use the left click as it
would interfere with the default behavior. But this is actually good
because currently the dragging can only be achieved with left click.

For the second thing, I think it was a typo in your code because I could
not find the reason for this. But I think it doesn't impact anything. Or
was it intentional? It is basically to reset it mouseDown right? Everywhere
else you use -1,-1.

On Mon, Jun 13, 2016 at 11:53 AM, Wenzel Jakob <notifications@github.com>
wrote:

> In src/textbox.cpp
> <https://github.com/wjakob/nanogui/pull/69#discussion_r66766255>:
>
> > @@ -250,7 +250,28 @@ bool TextBox::mouseButtonEvent(const Vector2i &p, int button, bool down,
> >                  /* Double-click: select all text */
> >                  mSelectionPos = 0;
> >                  mCursorPos = (int) mValueTemp.size();
> > -                mMouseDownPos = Vector2i(-1, 1);
>
> Is the sign change in the Y axis intentional here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wjakob/nanogui/pull/69/files/8ddcad132ca13bd4d8a6b0b9a6e29a03d7f24094#r66766255>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJhTYLiUAbSgmsITIGpAz1nGmlLKuanxks5qLSi1gaJpZM4IxX81>
> .
>